crates/sncast/src/starknet_commands/account/create.rs

```diff
@@ -379,7 +383,71 @@ fn generate_deploy_command_with_keystore(
     let network_flag = generate_network_flag(rpc_url, network);
 
     format!(
-        "\n\nAfter prefunding the address, run:\n\
+        "\n\nAfter prefunding the account, run:\n\
         sncast --account {account} --keystore {keystore} account deploy {network_flag}"
     )
 }
+
+fn felt_to_bigdecimal<F, D>(felt: F, decimals: D) -> BigDecimal
+where
+    F: AsRef<Felt>,
+    D: Into<i64>,
+{
+    BigDecimal::new(
+        BigInt::from_bytes_be(Sign::Plus, &felt.as_ref().to_bytes_be()),
+        decimals.into(),
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_felt_to_bigdecimal_with_zero() {
+        let felt = Felt::ZERO;
+        let result = felt_to_bigdecimal(felt, 18);
+        assert_eq!(result, BigDecimal::from(0));
+    }
+
+    #[test]
+    fn test_felt_to_bigdecimal_with_one() {
+        let felt = Felt::ONE;
+        let result = felt_to_bigdecimal(felt, 18);
+        assert_eq!(result, BigDecimal::new(BigInt::from(1), 18));
+    }
+
+    #[test]
+    fn test_felt_to_bigdecimal_with_large_number() {
+        let felt = Felt::from_hex("0x1234567890abcdef").unwrap();
+        let result = felt_to_bigdecimal(felt, 18);
+        let expected = BigDecimal::new(BigInt::from_bytes_be(Sign::Plus, &felt.to_bytes_be()), 18);
+        assert_eq!(result, expected);
```

Comment:
This isn't testing match though if we are using both values from the felt. Can't we create big decimal from some constant value we precompute?

Reply:
i.e. hardcode something